src/test/integration/sendgrid.test.ts

```diff
@@ -0,0 +1,56 @@
+import enmail from '../../index';
+import { mockApis } from '../utils/testMock';
+
+describe('Sendgrid adapter', () => {
+    const getSendgridAdapter = () => enmail.createAdapter.sendgrid(options);
+    const options = { apiKey: 'SG.apiKey' };
+    const adapter = getSendgridAdapter();
+    const sendOptions = { mailSettings: { sandboxMode: { enable: true } } };
+    const body = {
+        to: 'test@example.com',
+        from: 'test2@example.com',
+        content: 'Hello world',
+        subject: 'Hello',
+    };
+    beforeAll(() => mockApis());
+    it('Sendgrid adapter can be created', () => {
+        expect(typeof adapter.sender).toEqual('function');
+        expect(adapter.options).toStrictEqual(options);
+        expect(typeof adapter.mailService.send).toEqual('function');
+        expect(adapter).toMatchSnapshot();
+    });
+    it('Sender will be successfully returned', () => {
+        const sender = enmail.sender(adapter);
+        expect(typeof sender).toEqual('function');
+        expect(sender).toEqual(adapter.sender);
+    });
+    it('An email can be sent via adapter', async () => {
```

Comment:
This and the following tests just test the signature. It would be cool to test wether options get passed to call somehow (with jests' spy perhaps or module mock), that options overrides work etc. 200 is just nocked.